task/runner.go

```diff
@@ -221,6 +221,22 @@ func (r *runner) getAssetAndOS(assetID string) (*api.Asset, drivers.OSSession, e
 
 func (r *runner) publishTaskResult(ctx context.Context, task data.TaskInfo, output *data.TaskOutput, err error) error {
 	resultCh := make(chan event.PublishResult, 1)
+
+	if err != nil {
+		if strings.Contains(err.Error(), "unexpected EOF") {
+			err = errors.New("file download failed")
+		} else if strings.Contains(err.Error(), "error running ffprobe [] exit status 1") ||
+			strings.Contains(err.Error(), "Could not create stream ID") ||
+			strings.Contains(err.Error(), "502 Bad Gateway") ||
+			(strings.Contains(err.Error(), "EOF") && strings.Contains(err.Error(), "error processing file")) {
```

Comment:
Since this is the most verbose check here, I'd suggest you to leave it last (especially if you create a helper for humanizing these messages). I think it's a little easier to parse when there are only short and simple branches in the code and then 1 a little more complex one in the end. And if you have a helper, can even make early returns of the simple cases to make the complex part easier to digest e.g.
```go
func humanizeError(err error) error {
  if err == nil {
     return nil
  }
  errMsg := ...
  if errMsg ... {
     return errors.New(...)
  } else if
    // repeat a couple of these
 } 
   
 isProcessingError := strings.Contains(...) || ... && ...
 if isProcessingError {
    return errors.New(...)
 }
  return err
}
```

(actually, I think what would help the most there would be the helper variable, not only the fact that it is the last check. multi-line `if` conditions are very confusing IMO, even the default go indentation is weird. should try to avoid that whenever possible)

Reply:
Much cleaner <3